pkg/controller/node/nodecontroller.go

```diff
@@ -480,9 +480,26 @@ func (nc *NodeController) monitorNodeStatus() error {
 						continue
 					}
 					if remaining {
-						// queue eviction of the pods on the node
+						// Immediately evict pods (skip rate-limited evictor)
 						glog.V(2).Infof("Deleting node %s is delayed while pods are evicted", node.Name)
-						nc.evictPods(node.Name)
+						go func(nodeName string) {
+							nc.evictorLock.Lock()
+							defer nc.evictorLock.Unlock()
+							remaining, err := nc.deletePods(nodeName)
+							if err != nil {
+								glog.Errorf("Unable to evict pods from node %s: %v", nodeName, err)
+								nc.podEvictor.Add(nodeName)
+								return
+							}
+							if !remaining {
+								return
+							}
+							// Immediately terminate pods.
```

Comment:
That's what I had this morning, but @gmarek convinced me otherwise. We don't expect any *actual* pods to be running, though they may still be represented in the apiserver. If I understand the system, it should come out fine whether we queue or directly terminate. 

If we try to directly terminate and it fails for whatever reason, we'll bail, but next scan through the nodes we should attempt to evict/terminate again.

If we add to the evictor queue, we might get more retries before the next full node scan, but we may also be backing up the rate-limited piece for no good reason.

If that description doesn't match with what you think would actually happen, I'm happy to change the PR.